### PR TITLE
[7.1.0] Update Kafka-Connector doc with Avro Message support

### PR DESCRIPTION
**Pull request by Vathsan**
$subject
Related PR: wso2-extensions/esb-connector-kafka#55

### CODE REVIEW

File: en/micro-integrator/docs/references/connectors/kafka-connector/kafka-connector-overview.md [hunk 1]
```diff
@@ -20,19 +22,24 @@ For older versions, see the details in the connector store.
...
+* **[Kafka Connector Avro Message Producer Example](3.1.x/kafka-connector-avro-producer-example.md)**: This example demonstrates how to send Apache Avro messages to a Kafka broker via Kafka topics.  
+
+* **[Kafka Connector Reference](3.1.x/kafka-connector-config.md)**: This documentation provides a reference guide for the Kafka Connector.
 
 ## Kafka Inbound Endpoint documentation
 
 WSO2 EI Kafka inbound endpoint acts as a message consumer. It creates a connection to ZooKeeper and requests messages for a topic. The inbound endpoint is bundled with the Kafka connector.
 
-* **[Kafka Inbound Endpoint Example](kafka-inbound-endpoint-example.md)**: This sample demonstrates how one way message bridging from Kafka to HTTP can be done using the inbound Kafka endpoint. 
+* **[Kafka Inbound Endpoint Example](3.1.x/kafka-inbound-endpoint-example.md)**: This sample demonstrates how one way message bridging from Kafka to HTTP can be done using the inbound Kafka endpoint.
```
Comment (nilminiwso2) [inline en/micro-integrator/docs/references/connectors/kafka-connector/kafka-connector-overview.md:42]:
```suggestion
* **[Kafka Inbound Endpoint Example](3.1.x/kafka-inbound-endpoint-example.md)**: This sample demonstrates how one-way message bridging from Kafka to HTTP can be done using the inbound Kafka endpoint. 
```

File: en/micro-integrator/docs/references/connectors/kafka-connector/3.1.x/kafka-connector-producer-example.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
+# Kafka Connector Example
+
+Given below is a sample scenario that demonstrates how to send messages to a Kafka broker via Kafka topics. The publishMessages operation allows you to publish messages to the Kafka brokers via Kafka topics.
+
+## What you'll build
+
+Given below is a sample API that illustrates how you can connect to a Kakfa broker with the `init` operation and then use the `publishMessages` operation to publish messages via the topic. It exposes Kakfa functionalities as a RESTful service. Users can invoke the API using HTTP/HTTPs with the required information.
+
+API has the context `/publishMessages`. It will publish messages via the topic to the Kafka server.
```
Comment (nilminiwso2) [inline en/micro-integrator/docs/references/connectors/kafka-connector/3.1.x/kafka-connector-producer-example.md:9]:
```suggestion
API has the `/publishMessages` context. It publishes messages via the topic to the Kafka server.
```

File: en/micro-integrator/docs/references/connectors/kafka-connector/3.1.x/kafka-connector-producer-example.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+Before you begin, set up Kafka by following the instructions in [Setting up Kafka](setting-up-kafka.md).
+
+## Configure the connector in WSO2 Integration Studio
+
+Follow these steps to set up the Integration Project and the Connector Exporter Project.
+
+{!references/connectors/importing-connector-to-integration-studio.md!}
+
+1. Right click on the created Integration Project and select **New** -> **Rest API** to create the REST API.
```
Comment (nilminiwso2) [inline en/micro-integrator/docs/references/connectors/kafka-connector/3.1.x/kafka-connector-producer-example.md:27]:
```suggestion
1. Right-click the created Integration Project and select **New** -> **Rest API** to create the REST API.
```

File: en/micro-integrator/docs/references/connectors/kafka-connector/3.0.x/setting-up-kafka.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
+# Setting up Kafka
+
+To use the Kafka connector, download and install [Apache Kafka](http://kafka.apache.org/downloads.html). Before you start configuring the Kafka you also need WSO2 EI and we refer to that location as <PRODUCT_HOME>.
```
Comment (nilminiwso2) [inline en/micro-integrator/docs/references/connectors/kafka-connector/3.0.x/setting-up-kafka.md:3]:
```suggestion
To use the Kafka connector, download and install [Apache Kafka](http://kafka.apache.org/downloads.html). Before you start configuring Kafka, you also need WSO2 EI and we refer to that location as <PRODUCT_HOME>.
```

File: en/micro-integrator/docs/references/connectors/kafka-connector/3.1.x/kafka-connector-avro-producer-example.md [hunk 5]
```diff
@@ -0,0 +1,173 @@
+# Avro Message with Kafka Connector Example
+
+Given below is a sample scenario that demonstrates how to send Apache Avro messages to a Kafka broker via Kafka topics. The publishMessages operation allows you to publish messages to the Kafka brokers via Kafka topics.
+
+## What you'll build
+
+Given below is a sample API that illustrates how you can connect to a Kakfa broker with the `init` operation and then use the `publishMessages` operation to publish messages via the topic. It exposes Kakfa functionalities as a RESTful service. Users can invoke the API using HTTP/HTTPs with the required information.
```
Comment (nilminiwso2) [inline en/micro-integrator/docs/references/connectors/kafka-connector/3.1.x/kafka-connector-avro-producer-example.md:7]:
```suggestion
Given below is a sample API that illustrates how you can connect to a Kafkaa broker with the `init` operation and then use the `publishMessages` operation to publish messages via the topic. It exposes Kakfa functionalities as a RESTful service. Users can invoke the API using HTTP/HTTPs with the required information.
```

File: en/micro-integrator/docs/references/connectors/kafka-connector/3.1.x/kafka-connector-avro-producer-example.md [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+
+API has the context `/publishMessages`. It will publish messages via the topic to the Kafka server.
```
Comment (nilminiwso2) [inline en/micro-integrator/docs/references/connectors/kafka-connector/3.1.x/kafka-connector-avro-producer-example.md:9]:
```suggestion
API has the `/publishMessages` context. It publishes messages via the topic to the Kafka server.
```

File: en/micro-integrator/docs/references/connectors/kafka-connector/3.1.x/kafka-connector-avro-producer-example.md [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+   
+Run the following command to verify the messages:
+````bash
+[confluent_home]/bin/kafka-avro-console-consumer.sh --topic myTopic --bootstrap-server localhost:9092 --property print.key=true --from-beginning
+````
+See the following message content:
+````json
+{"f1":{"string":"sampleValue"}}
+````  
+Sample API configuration when Confluent Schema Registry is secured with basic auth,
```
Comment (nilminiwso2) [inline en/micro-integrator/docs/references/connectors/kafka-connector/3.1.x/kafka-connector-avro-producer-example.md:110]:
```suggestion
Sample API configuration when the Confluent Schema Registry is secured with basic auth,
```

File: en/micro-integrator/docs/references/connectors/kafka-connector/3.1.x/kafka-connector-avro-producer-example.md [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+               <value>{$ctx:value}</value>
+               <valueSchema>{$ctx:valueSchema}</valueSchema>
+            </kafkaTransport.publishMessages>
+        </inSequence>
+        <outSequence/>
+        <faultSequence/>
+    </resource>
+</api>
+```
+In the above example the <b>basicAuthCredentialsSource</b> parameter is configured to <b>USER_INFO</b>. If <b>basicAuthCredentialsSource</b> parameter is set to URL as,
```
Comment (nilminiwso2) [inline en/micro-integrator/docs/references/connectors/kafka-connector/3.1.x/kafka-connector-avro-producer-example.md:155]:
```suggestion
In the above example, the <b>basicAuthCredentialsSource</b> parameter is configured as <b>USER_INFO</b>. For example, consider a scenario where the <b>basicAuthCredentialsSource</b> parameter is set to <b>URL</b> as follows:
```

File: en/micro-integrator/docs/references/connectors/kafka-connector/3.1.x/kafka-connector-avro-producer-example.md [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+
+````xml 
+<basicAuthCredentialsSource>URL</basicAuthCredentialsSource>
+````
+
+then the <b>schemaRegistryUrl</b> parameter should be configured as shown below,
```
Comment (nilminiwso2) [inline en/micro-integrator/docs/references/connectors/kafka-connector/3.1.x/kafka-connector-avro-producer-example.md:161]:
```suggestion
Then, the <b>schemaRegistryUrl</b> parameter should be configured as shown below.
```

File: en/micro-integrator/docs/references/connectors/kafka-connector/3.1.x/kafka-connector-avro-producer-example.md [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+
+````xml 
+<schemaRegistryUrl>http://admin:admin@localhost:8081</schemaRegistryUrl>
+````
+You may refer to the [confluent documentation](https://docs.confluent.io/platform/current/schema-registry/serdes-develop/serdes-avro.html) for more details.
+
+This demonstrates how the Kafka connector publishes avro messages to the Kafka brokers.
```
Comment (nilminiwso2) [inline en/micro-integrator/docs/references/connectors/kafka-connector/3.1.x/kafka-connector-avro-producer-example.md:168]:
```suggestion
This demonstrates how the Kafka connector publishes Avro messages to Kafka brokers.
```

File: en/micro-integrator/docs/references/connectors/kafka-connector/3.1.x/kafka-connector-avro-producer-example.md (hunk 5 above)
Comment (Vathsan) [reply en/micro-integrator/docs/references/connectors/kafka-connector/3.1.x/kafka-connector-avro-producer-example.md:7]: Should not this be Kafka?

File: en/micro-integrator/docs/references/connectors/kafka-connector/3.0.x/kafka-connector-producer-example.md [hunk 11]
```diff
@@ -0,0 +1,108 @@
+# Kafka Connector Example
+
+Given below is a sample scenario that demonstrates how to send messages to a Kafka broker via Kafka topics. The publishMessages operation allows you to publish messages to the Kafka brokers via Kafka topics.
```
Comment (nilminiwso2) [inline en/micro-integrator/docs/references/connectors/kafka-connector/3.0.x/kafka-connector-producer-example.md:3]:
```suggestion
Given below is a sample scenario that demonstrates how to send messages to a Kafka broker via Kafka topics. The `publishMessages` operation allows you to publish messages to the Kafka brokers via Kafka topics.
```

File: en/micro-integrator/docs/references/connectors/kafka-connector/3.0.x/kafka-connector-producer-example.md [hunk 12]
```diff
@@ -0,0 +1,108 @@
+# Kafka Connector Example
+
+Given below is a sample scenario that demonstrates how to send messages to a Kafka broker via Kafka topics. The `publishMessages` operation allows you to publish messages to the Kafka brokers via Kafka topics.
+
+## What you'll build
+
+Given below is a sample API that illustrates how you can connect to a Kakfa broker with the `init` operation and then use the `publishMessages` operation to publish messages via the topic. It exposes Kakfa functionalities as a RESTful service. Users can invoke the API using HTTP/HTTPs with the required information.
+
+API has the context `/publishMessages`. It will publish messages via the topic to the Kafka server.
```
Comment (nilminiwso2) [inline en/micro-integrator/docs/references/connectors/kafka-connector/3.0.x/kafka-connector-producer-example.md:9]:
```suggestion
API has the context `/publishMessages`. It publishes messages via the topic to the Kafka server.
```

**Review by nilminiwso2 — APPROVED** [review]

**Pull request merged**